parquet-hadoop/src/main/java/org/apache/parquet/hadoop/metadata/KeyValueMetadataMergeStrategy.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.hadoop.metadata;
+
+import org.apache.parquet.schema.MessageType;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Set;
+
+import static java.util.Collections.unmodifiableMap;
+
+/**
+ * Strategy to merge metadata. Possible implementations:
+ * 1) Expect metadata to be exactly identical. Otherwise throw error (this is the default strategy implementation)
+ * 2) Custom strategy to merge metadata if the values are not identical. Example: concatenate values.
+ */
+public interface KeyValueMetadataMergeStrategy extends Serializable {
+  /**
+   * @param keyValueMetaData the merged app specific metadata
+   *
+   * @throws NullPointerException if keyValueMetaData is {@code null}
+   */
+  Map<String, String> merge(Map<String, Set<String>> keyValueMetaData);
```

Comment:
I think, this interface does not support all types of the potential merge strategy implementations. For example if you would like to use the value which is common in the most files you cannot do it because you only have the distinct values. What do you think about using a `List<String>` instead of a `Set<String>`?

Reply:
GlobalMetadata uses Map<String, Set<String>> https://github.com/apache/parquet-mr/blob/master/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/metadata/GlobalMetaData.java#L41. So merge method takes in same argument.

Do you think it makes sense to change GlobalMetadata to track items in List<String>? I'm not sure if it would be backward compatible and if there are use cases expecting this to be Set.  

Without changing GlobalMetadata#keyValueMetaData, it may not be useful to change this interface. Let me know what you think.

Reply:
People may already rely on the 'set' to do the dedup.

Reply:
Sorry, I've missed this type is already in our API. Agreed to keep the interface as is.